src/components/RasterPreview.js

```diff
@@ -2,134 +2,171 @@
-            />
-          </p>) : null}
-
-        {chooseLocation ? (
-          <SelectAsset
-            placeholderText={placeholderAssetSelection}
-            setAsset={this.handleSetAsset}
-            spatialBounds={raster ? raster.spatial_bounds : null}
-            setLocation={setLocation}
-          />) : null}
-
-        <Map
-          onClick={this.handleMapClick.bind(this)}
-          className={styles.MapStyle}
-          {...mapLocation}
+      ]
+    };
+  } else {
+    mapLocation = {
+      center: marker,
+      zoom: 8
+    };
+  };
+
+  return (
+    <div
+      className={styles.MapContainer}
+    >
+      {chooseLocation ?
+        <div
+          className={styles.AssetSelect}
         >
-          <TileLayer
-            url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
-            attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
-            setLocation={this.props.setLocation}
-          />
-          {raster ? (
-            <WMSTileLayer
-              url="/wms/"
-              styles={this.formatWMSStyles(raster.options.styles)}
-              layers={this.formatWMSLayers(raster.wms_info.layer)}
-              opacity={0.9}
-            />) : null}
-          <TileLayer
-            url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
-            attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+          <SelectDropdown
+            title={''}
+            name={name}
+            placeholder={'- Search and select an asset -'}
+            value={selectedAsset}
+            valueChanged={value => {
+              setSelectedAsset(value);
+              if (value) {
+                setLocation(value.value);
+              } else {
+                setLocation(null);
+              };
+            }}
+            options={[]}
+            validated={validated}
+            errorMessage={errorMessage}
+            triedToSubmit={triedToSubmit}
+            isAsync
+            loadOptions={searchInput => fetchAssets(raster, searchInput)}
           />
-          {location ?
-           <Marker position={marker} />
-          : null}
-        </Map>
-      </div>
-    );
-  }
+        </div>
+      : null}
+      <Map
+        onClick={handleMapClick}
+        className={styles.MapStyle}
+        zoomControl={false}
+        {...mapLocation}
+      >
+        <ZoomControl position="bottomright"/>
+        <TileLayer
+          url={`https://api.mapbox.com/styles/v1/nelenschuurmans/ck8sgpk8h25ql1io2ccnueuj6/tiles/256/{z}/{x}/{y}@2x?access_token=${mapBoxAccesToken}`}
+          attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+          setLocation={setLocation}
```

Comment:
I do not understand this setLocation.
Maybe I am missing something, but I can not find it as part of the api here:
https://react-leaflet.js.org/docs/api-components

Reply:
Ah you are right. I didn't notice it. Kept it from the old code. It didn't do anything so I removed it. Also removed the attribution since it's not correct that we are using OpenStreetMap.